employees/tests/test_unit_report_viewset.py

```diff
@@ -148,3 +151,312 @@ def test_report_detail_view_should_delete_report_on_delete(self):
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': new_description,
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+                'discard': "Discard"
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        # SELF NOTE: It appears that save() does not override reference on model object (edit is unavailable)
+        self.assertEqual(response.status_code, 302)
+        self.assertNotEqual(current_description, new_description)
+
+    def test_custom_report_detail_view_should_not_update_report_on_post_if_form_is_invalid(self):
+        new_description = 'Some other description'
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'description': new_description,
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        # SELF NOTE: It appears that save() does not override reference on model object (edit is unavailable)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNotNone(response.data['errors'])
+        self.assertNotEqual(new_description, current_description)
+
+
+class DeleteReportTests(TestCase):
+    def setUp(self):
+        self.user = CustomUser(
+            email="testuser@codepoets.it",
+            password='newuserpasswd',
+            first_name='John',
+            last_name='Doe',
+            country='PL'
+        )
+        self.user.full_clean()
+        self.user.save()
+
+        self.project = Project(
+            name="Test Project",
+            start_date=datetime.datetime.now(),
+        )
+        self.project.full_clean()
+        self.project.save()
+
+        self.report = Report(
+            date=datetime.datetime.now().date(),
+            description='Some description',
+            author=self.user,
+            project=self.project,
+            work_hours=Decimal('8.00'),
+        )
+        self.report.full_clean()
+        self.report.save()
+
+    def test_delete_report_view_should_delete_report_on_post(self):
+        request = APIRequestFactory().delete(path=reverse('custom-report-delete', args=(self.report.pk,)))
+        request.user = self.user
+        response = delete_report(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 0)
```

Comment:
There is missing test for that user A can't delete user's B report.

Reply:
Except he can, until those permissions are introduced. I wouldn't add tests that are meant to fail. There are separate branches with permissions written for entire app introduced later on.